CSS fallbacks for home page

singingwolfboy opened the pull request:
Fixes #1559. Changes the CSS so that course cards tile correctly, even without course images and with variable course description lengths.
![home-page-tiles](https://cloud.githubusercontent.com/assets/132355/20361218/c7f3a664-ac04-11e6-8fe6-46b6f77f9c53.png)

amir-qayyum-khan commented:
there are issues with UI when program description is large, can you look into this.

<img width="1280" alt="screen shot 2016-11-17 at 5 09 37 pm" src="https://cloud.githubusercontent.com/assets/10431250/20388868/ab4804b2-ace8-11e6-9325-3956ad882c47.png">

amir-qayyum-khan commented:
issue with image title

![imageedit_2_9197766907](https://cloud.githubusercontent.com/assets/10431250/20389038/6bf6d558-ace9-11e6-9ac1-b3ee2f071fcf.gif)

singingwolfboy commented:
When the program description is very long, it becomes scrollable -- if you hover your mouse over the description, you can scroll down. I set a defined height for the description text, to ensure that all the course cards are the same height, which is necessary for the course cards to tile correctly.

The only other way I can think of to make this work properly is to use the [CSS Flexbox layout](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout/Using_CSS_flexible_boxes), which might make this work without requiring scrollable description text. However, I'm not very familiar with that. @roberthouse54, maybe you can help out?

Also, I'm not sure what the problem is with the image title. When the title is too wide, it will wrap to the next line. How should it behave instead?

roberthouse54 commented:
I think the solution may be to set the height of the div (as done here) and then set a character limit on the program description in the CMS. About the course title wrapping... let me think a bit more on that, but please just add an issue for that and assign it to me.

amir-qayyum-khan commented:
@singingwolfboy there is one issue , with big height the space between 2 program rows gets 0,

please see 
<img width="1280" alt="screen shot 2016-11-17 at 5 09 37 pm" src="https://cloud.githubusercontent.com/assets/10431250/20388868/ab4804b2-ace8-11e6-9325-3956ad882c47.png">

singingwolfboy commented:
I've changed the CSS so that the "More about..." link is always shown, even when the description is long enough that it will scroll. I also added some space between rows. Does that look better?
![screen shot 2016-11-17 at 12 06 26 pm](https://cloud.githubusercontent.com/assets/132355/20399288/5dc24150-acbe-11e6-9a19-1db1358fd5c5.png)

roberthouse54 commented:
Looks good, but can you please increase the space more? Like this: 
![image](https://cloud.githubusercontent.com/assets/20047260/20402231/db97552e-acc9-11e6-98da-56a68d0b8580.png)

singingwolfboy commented:
Rebased and updated. How does this look?

![bacon-ipsum](https://cloud.githubusercontent.com/assets/132355/20671739/9703aa12-b54c-11e6-930c-e815d1225062.png)

amir-qayyum-khan commented:
@singingwolfboy please reposition 

<img width="319" alt="screen shot 2016-11-28 at 7 57 03 pm" src="https://cloud.githubusercontent.com/assets/10431250/20672943/dfadf9a4-b5a4-11e6-8f82-c485ec0fa3db.png">

**according to this.**
https://cloud.githubusercontent.com/assets/20047260/19897923/66ae314c-a030-11e6-9201-d5f0bf54cc71.png

singingwolfboy commented:
@amir-qayyum-khan That's outside the scope of this pull request. This pull request is _only_ about adjusting the course cards CSS so that they tile correctly. The mailing list button can be repositioned in a separate pull request.

amir-qayyum-khan commented:
then good to go. 👍

noisecapella commented:
@singingwolfboy please delete your branch after you merge the PR
